docs/content/any/guides/data_filtering/_index.md

```diff
@@ -178,10 +282,41 @@ def get_repos(constraints):
 
 {{% /ifLang %}}
 
-When you register classes you need to specify two new things. One is `types` which is a map that says what the type of each field in the class is. This can be base types like `String` or another registered class or it can be a `Relationship`.
+{{% ifLang "ruby" %}}
 
-The Relationship specifies the type of the related value. It also says how it's related. Any instance of the other type that has their `other_field` equal to the current objects `my_field` is related. `kind` can be either `"parent"` which means there is one related instance, or `"children"` which means there are many.
-Internally the fields are used to create constraints and the related instances are fetched using the fetching functions.
+```ruby
+# A module to autogenerate a fetcher function for an ActiveRecord subclass.
+# `include` it in `Class`and the fetcher is accessible in `Class::FETCHER`
+module ActiveRecordFetcher
+  def self.included(base)
+    base.class_eval do
+      kinds = Hash.new { |k| raise "Unsupported constraint kind: #{k}" }
+      kinds['Eq'] = kinds['In'] = lambda do |q, c|
+        q.where(
+          if c.field.nil?
+            { primary_key => c.value.send(primary_key) }
+          else
+            { c.field => c.value }
+          end
+        )
+      end
+
+      const_set(:FETCHER, lambda do |cons|
+        cons.reduce(self) { |q, con| kinds[con.kind][q, con] }
+      end)
+    end
+  end
+end
```

Comment:
Hmm, I don't really write much ruby so I dunno if this is idiomatic and obvious to ruby people but for me it was maybe a little too "clever".

Could we maybe just have it be one function and put the logic into the block of the reduce instead of a lambda for each kind (especially since they do the same thing) and then just set FETCHER to that function?

Also why do `Eq` and `In` do the same thing, is that an activerecord thing where a list automatically means `In`?

Reply:
ok no problem & yes it's an active record thing